builtin/logical/pki/cert_util.go

```diff
@@ -185,32 +185,72 @@ func fetchCAInfo(req *logical.Request) (*caInfoBundle, error) {
 // separate pathing for CA, CRL, and revoked certificates.
 func fetchCertBySerial(req *logical.Request, prefix, serial string) (*logical.StorageEntry, error) {
 	var path string
+	var err error
+	var certEntry *logical.StorageEntry
+
+	hyphenSerial := strings.Replace(strings.ToLower(serial), ":", "-", -1)
+	colonSerial := strings.Replace(strings.ToLower(serial), "-", ":", -1)
 
 	switch {
 	// Revoked goes first as otherwise ca/crl get hardcoded paths which fail if
 	// we actually want revocation info
 	case strings.HasPrefix(prefix, "revoked/"):
-		path = "revoked/" + strings.Replace(strings.ToLower(serial), "-", ":", -1)
+		path = "revoked/" + hyphenSerial
 	case serial == "ca":
 		path = "ca"
 	case serial == "crl":
 		path = "crl"
 	default:
-		path = "certs/" + strings.Replace(strings.ToLower(serial), "-", ":", -1)
+		path = "certs/" + hyphenSerial
 	}
 
-	certEntry, err := req.Storage.Get(path)
+	certEntry, err = req.Storage.Get(path)
 	if err != nil {
 		return nil, errutil.InternalError{Err: fmt.Sprintf("error fetching certificate %s: %s", serial, err)}
 	}
-	if certEntry == nil {
+	if certEntry != nil {
+		if certEntry.Value == nil || len(certEntry.Value) == 0 {
+			return nil, errutil.InternalError{Err: fmt.Sprintf("returned certificate bytes for serial %s were empty", serial)}
+		}
+		return certEntry, nil
+	}
+
+	// No point checking these, no old/new style colons/hyphens
+	if path == "ca" || path == "crl" {
 		return nil, nil
 	}
 
+	// Save the desired path
+	desiredPath := path
```

Comment:
Oh nevermind.  That does not work.  That was also one of the reason I wanted to move away from the multiple replacements...  Maybe it makes sense to create the `legacyPath` and `path` instead of just tracking the serial.